src/frontier_exploration/scripts/frontier_filter.py

```diff
@@ -90,6 +95,49 @@ def init_markers(self):
 
         return points_clust
 
+    def is_valid_frontier(self, node):
+        if node[0] > -0.5 and node[0] < 12.0 and node[1] > -10.0 and node[1] < 2.0:
+            return True
+        return False
+
+    def obstacle_free(self, xnear, xnew):
```

Comment:
Can we rename this function? Obstacle free doesnt reflect what this function is doing at all, can we rename it to something like collision_check_connection